### PR TITLE
Add ScatterUpdate value infer

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *If ScatterUpdate appear in IR on shape route it will not calculate value and resulting shape will be determined as dynamic. This will result in failure of MO IR Reader when it will verify shapes after read.*
 - *This issue appeared only after ScatterElementsUpdate to ScatterUpdate transformation was enabled for offline step.*

### Tickets:
 - *89041*

### CODE REVIEW

**Comment by akladiev** [comment]
The valid link to the validation results: https://openvino-ci.toolbox.iotg.sclab.intel.com/job/private-ci/job/github_trigger/job/openvino/job/PR-12595/1/

File: tools/mo/unit_tests/mo/ops/scatter_test.py [hunk 1]
```diff
@@ -101,3 +100,70 @@ def test_scatterelements_value_infer(self, data, indices, updates, axis, ref_res
...
         self.assertTrue(np.array_equal(ref_res, res_output_value))
+
+
+@generator
+class ScatterUpdateInferTest(unittest.TestCase):
+    @generate(*[
+        ([[0.0, 0.0, 0.0],
+          [0.0, 0.0, 0.0],
+          [0.0, 0.0, 0.0]],
+         [[1, 2]],
```
Comment (rkazants) [inline tools/mo/unit_tests/mo/ops/scatter_test.py:111]: can you please add cases with the first dimension of indices bigger that one?

File: tools/mo/unit_tests/mo/ops/scatter_test.py [hunk 2]
```diff
@@ -101,3 +100,70 @@ def test_scatterelements_value_infer(self, data, indices, updates, axis, ref_res
...
+         [[[1.0, 1.1, 1.2],
+           [2.0, 2.1, 2.2]]],
+         0,
+         [[0.0, 0.0, 0.0],
+          [1.0, 1.1, 1.2],
+          [2.0, 2.1, 2.2]]),
+
+        # negative axis
```
Comment (rkazants) [inline tools/mo/unit_tests/mo/ops/scatter_test.py:119]: can you please add a cases with the following shapes:
```
data_shape = [2, 3, 3]
indices_shape = [3, 2]
updates shape = [2, 3, 2, 3]
axis = 1
```

Comment (mvafin) [reply tools/mo/unit_tests/mo/ops/scatter_test.py:119]: Done

**Review by rkazants — APPROVED** [review]

**Pull request merged**